File: cli-development/templates/nodejs-commander/src/commands/delete.ts
import { Command } from 'commander'
import inquirer from 'inquirer'
import chalk from 'chalk'
import ora from 'ora'

interface DeleteOptions {
  force?: boolean
}

export function deleteCommand(): Command {
  return new Command('delete')
    .description('Delete a project')
    .argument('<name>', 'Project name')
    .option('-f, --force', 'Force delete without confirmation')
    .action(async (name: string, options: DeleteOptions) => {
      try {
        await deleteProject(name, options)
      } catch (error) {
        console.error(chalk.red('✗ Failed to delete project'))
        console.error(error)
        process.exit(1)
      }
    })
}

async function deleteProject(
  name: string,
  options: DeleteOptions
): Promise<void> {
  if (!options.force) {
    const { confirmed } = await inquirer.prompt([
      {
        type: 'confirm',
        name: 'confirmed',
        message: `Delete project "${name}"?`,
        default: false
      }
    ])

    if (!confirmed) {
      console.log(chalk.yellow('Cancelled'))
      return
    }
  }

  const spinner = ora(`Deleting project: ${name}`).start()

  try {
    // シミュレーション
    await new Promise(resolve => setTimeout(resolve, 1000))

    spinner.succeed(chalk.green('Project deleted'))
  } catch (error) {
    spinner.fail(chalk.red('Failed to delete project'))
    throw error
  }
}

File: cli-development/templates/python-typer/src/cli/commands/delete.py
"""Delete command."""
import typer
from rich.console import Console
import time

console = Console()

def delete_project(
    name: str = typer.Argument(..., help="Project name"),
    force: bool = typer.Option(False, "--force", "-f", help="Force delete without confirmation"),
) -> None:
    """Delete a project."""
    if not force:
        confirmed = typer.confirm(f'Delete project "{name}"?')
        if not confirmed:
            console.print("[yellow]Cancelled[/yellow]")
            raise typer.Abort()

    console.print(f"[red]Deleting project: {name}[/red]")
    time.sleep(1)
    console.print("[green]✓ Project deleted[/green]")
